wrapanapi/containers/providers/rhopenshift.py

```diff
@@ -0,0 +1,892 @@
+                continue
+            _, name, image_id, _ = Image.parse_docker_image_info(entity_j['dockerImageReference'])
+            entities.append(Image(self, name, image_id))
+        return entities
+
+    def list_deployment_config(self):
+        """Returns list of deployment configs"""
+        entities = []
+        entities_j = self.o_api.get('deploymentconfig')[1]['items']
+        for entity_j in entities_j:
+            meta = entity_j['metadata']
+            entity = DeploymentConfig(self, meta['name'], meta['namespace'])
+            entities.append(entity)
+        return entities
+
+    def deploy_template(self, template, tags=None, password='smartvm', **kwargs):
+        """Deploy a VM from a template
+
+        Args:
+            template: (str) The name of the template to deploy
+            tags: (dict) dict with tags if some tag isn't passed it is set to 'latest'
+            password: this password will be set as default everywhere
+        Returns: dict with parameters necessary for appliance setup or None if deployment failed
+        """
+        self.logger.info("starting template %s deployment", template)
+        self.wait_template_exist(namespace=self.default_namespace, name=template)
+
+        if not self.base_url:
+            raise ValueError("base url isn't provided")
+
+        prepared_tags = {key: 'latest' for key in self.template_tags}
+        if tags:
+            not_found_tags = [tag for tag in tags.keys() if tag not in self.stream_tags]
+            if not_found_tags:
+                raise ValueError("Some passed tags {t} don't exist".format(t=not_found_tags))
+            for tag, value in tags.items():
+                prepared_tags[self.stream2template_tags_mapping[tag]] = value
+
+        # create project
+        # assuming this is cfme installation and generating project name
+        proj_id = "".join(choice(string.digits + string.lowercase) for _ in range(6))
+        proj_name = "{t}-project-{pid}".format(pid=proj_id, t=template)
+        proj_url = "{proj}.{base_url}".format(proj=proj_id, base_url=self.base_url)
+        self.logger.info("unique id %s, project name %s", proj_id, proj_name)
+        self.create_project(name=proj_name)
+
+        # grant rights according to scc
+        self.logger.info("granting rights to project %s sa", proj_name)
+        scc_user_mapping = (
+            {'scc': 'anyuid', 'user': 'cfme-anyuid'},
+            {'scc': 'anyuid', 'user': 'cfme-orchestrator'},
+            {'scc': 'anyuid', 'user': 'cfme-httpd'},
+            {'scc': 'privileged', 'user': 'cfme-privileged'},
+        )
+
+        self.logger.info("granting required rights to project's service accounts")
+        security_api = self.ociclient.SecurityOpenshiftIoV1Api()
+        for mapping in scc_user_mapping:
+            old_scc = security_api.read_security_context_constraints(name=mapping['scc'])
+            got_users = old_scc.users if old_scc.users else []
```

Comment:
Is this done because `old_scc.users` could be `None`?  If so, can't this be:

```
got_users = old_scc.users or []

Reply:
yes, python-openshift-client sets variables to None when those are empty